modules/example/src/main/scala/example/Main.scala

```diff
@@ -45,14 +45,14 @@ object Main extends ZIOAppDefault {
     val deleteIndex: RIO[ElasticExecutor, Unit] =
       for {
         _ <- ZIO.logInfo(s"Deleting index '$Index'...")
-        _ <- ElasticRequest.deleteIndex(Index).execute
+        _ <- ZIO.serviceWithZIO[ElasticExecutor](_.execute(ElasticRequest.deleteIndex(Index)))
```

Comment:
This one is not bad, though it might need to adjust types to make it fully type-safe and maximize ergonimics.